### ISSUE TITLE
TileLayer sometimes loads multiple times during zoom animation

### ISSUE THREAD

**Issue by gagamilano** [Leaflet/Leaflet#8073]
- [x] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
- Create a tile layer
- Set events for zoomstart, zoomend, loading and load and add a simple console.log for each
- Play with zoom
Sometimes the load event only starts once: [screenshot](https://i.ibb.co/h95Wn6Y/Immagine-2022-03-17-120500.png)
Some other times it fires twice during the same animation: [screenshot](https://i.ibb.co/HNKWKpD/Immagine-2022-03-17-120310.png)
Note: these were all 1-level zoom changes.
- Now, add zoomAnimation = false when initializing the map. The tile loads once every time

**Expected behavior**
The tile should only be loading once, like it does with no animations

**Current behavior**
The tile is sometimes loading multiple times during the same zoom animation

**Environment**
- Leaflet version: 1.7.1
- Browser (with version): Versione 99.0.4844.74, Firefox 98.0.1
- OS/Platform (with version): Windows 10

**Additional context**
The feature I'm trying to work on relies on the tile's load event to load features onto the map. Every time the tile loads I have to remove the previous ones and load a new set of features. But with the animations turned on and the tiles loading twice in a row I get a lot of flickering. The only way I've found to make sure my features are loaded only once is disabling zoom animations.
I found the zoomAnimation workaround browsing other issues. See #7234 , except in my case I didn't need to turn off fade animations as well.

**Minimal example reproducing the issue**

https://plnkr.co/edit/ENeOvh40r1t3BXkL

- [x] this example is as simple as possible
- [x] this example does not rely on any third party code

**Comment by Malvoz** [comment]
Related / possibly duplicate of https://github.com/Leaflet/Leaflet/issues/4623, https://github.com/Leaflet/Leaflet/issues/4109.

**Comment by Falke-Design** [comment]
> Some other times it fires twice during the same animation: 

Only happens while zooming out
